python_gdeploy/wrapper/gdeploy_wrapper.py

```diff
@@ -0,0 +1,48 @@
+import os
+import subprocess
+import uuid
+
+GDEPLOY_CONFIG_PATH = "/var/run/python_gdeploy/config_"
+
+
+def add_section(section):
+    return "[" + section + "]\n"
+
+
+def cook_gdeploy_config(recipe):
+    config_str = ""
+    for el in recipe:
+        k = el.keys()[0]
+        v = el.values()[0]
+        config_str += add_section(k)
+        if type(v) == list:
+            for el in v:
+                config_str += el + "\n"
+        elif type(v) == dict:
+            for var, val in v.iteritems():
+                if val is None:
+                    line = val
+                    continue
+                line = var + "=" + (
+                    str(val) if type(val) != list else ",".join(val))
+                config_str += line + "\n"
+        config_str += "\n"
+
+    return config_str
+
+
+def invoke_gdeploy(config):
```

Comment:
Yes, this is thread safe as: we are having a uuid appended at the end of each conf file, each call to this function will have its own conf file. Also in the backend "gdeploy -c" can handle multiple simultaneous calls without any issue.